### ISSUE TITLE
Intensity Filter Breaks with Big Freaking intensities

### ISSUE THREAD

**Issue by DLu** [ros-perception/laser_filters#8]
If intensities in a laser scan are large enough, intensity filter will segfault. 

Given an intensity of 3e20 (for example), so [this line](https://github.com/ros-perception/laser_filters/blob/groovy-devel/include/laser_filters/intensity_filter.h#L96) results in an integer so large that it overflows. However the negative index results in the segfault [here](https://github.com/ros-perception/laser_filters/blob/groovy-devel/include/laser_filters/intensity_filter.h#L99)

Corrected version:

```
  int cur_bucket = (int) ((filtered_scan.intensities[i]/hist_max)*num_buckets) ;
  if (cur_bucket >= num_buckets-1 || cur_bucket < 0)
    cur_bucket = num_buckets-1 ;

  histogram[cur_bucket]++ ;
```

**Comment by mintar** [comment]
Hi David, I've chosen a slightly different approach in my pull request, so we don't have to rely on the integer overflow producing a negative number.

**Comment by DLu** [comment]
Good call. Thanks!

**Comment by mintar** [comment]
Wow, that was quick! :-)

**Comment by vrabaud** [comment]
Sorry for the delay, GSOC keeps us busy these days :)
The histogram is actually for debug purposes only. @mintar patch was modifying the filtered_scan. Is that a desired behavior ? I did that instead: 2da0e525b1bd2c58a5269901eb57325fece1cd9e

Please review and close the bug appropriately.

**Comment by mintar** [comment]
@vrabaud Thanks, this works. I didn't care about modifying the filtered_scan, because the relevant points are removed anyway by the line before (by setting the range to max + 1), so the intensities that I modified should be ignored anyway. That behaviour isn't "desired", it shouldn't matter one way or the other.

I don't have the permissions to close this issue, but I just verified it works now, so feel free to close the issue.
